ui_testing/testcases/basic_tests/test04_testunits.py

```diff
@@ -950,82 +951,38 @@ def test_027_changing_testunit_type_update_fields_accordingly(self, testunit_typ
-        self.assertEqual(testunits_request['status'], 1)
-        testunits = testunits_request['testUnits']
-        self.assertNotEqual(len(testunits), 0)
-        testunit_name = ''
-        for testunit in testunits:
-            if testunit['specifications'] == '' and testunit['quantification'] != '':
-                testunit_name = testunit['name']
-                break
-
-        if testunit_name == '':
-            self.base_selenium.LOGGER.info('there is no testunit with quantification only value')
-            self.assertTrue(False)
-
-        testunit_record = self.test_unit_page.search(value=testunit_name)[0]
-
-        self.test_unit_page.open_edit_page(row=testunit_record)
-
-        self.base_selenium.LOGGER.info('generate random lower/ upper limit')
+        self.info('generate random lower/ upper limit')
         random_lower_limit = self.test_unit_page.generate_random_number(lower=0, upper=49)
         random_upper_limit = self.test_unit_page.generate_random_number(lower=50, upper=100)
-
-        self.base_selenium.LOGGER.info('switch to specification')
-        self.test_unit_page.switch_from_quan_to_spec(lower_limit=random_lower_limit, upper_limit=random_upper_limit)
-        self.base_selenium.LOGGER.info('refresh to make sure that data are updated successfully')
-        self.base_selenium.refresh()
-
-        self.assertEqual(self.test_unit_page.get_testunit_specification_type(), 'spec')
-        self.assertEqual(self.test_unit_page.get_spec_upper_limit(), str(random_upper_limit))
-        self.assertEqual(self.test_unit_page.get_spec_lower_limit(), str(random_lower_limit))
+        if spec_or_quan == 'spec':
+            self.info('switch to quantification')
+            self.test_unit_page.switch_from_spec_to_quan(lower_limit=random_lower_limit,
+                                                         upper_limit=random_upper_limit)
+            self.info('refresh to make sure that data are updated successfully')
+            self.base_selenium.refresh()
+            self.assertEqual(self.test_unit_page.get_testunit_specification_type(), 'quan')
+            self.assertEqual(self.test_unit_page.get_quan_upper_limit(), str(random_upper_limit))
+            self.assertEqual(self.test_unit_page.get_quan_lower_limit(), str(random_lower_limit))
+        else:
+            self.info('switch to specification')
+            self.test_unit_page.switch_from_quan_to_spec(lower_limit=random_lower_limit,
+                                                         upper_limit=random_upper_limit)
+            self.info('refresh to make sure that data are updated successfully')
+            self.base_selenium.refresh()
+            self.assertEqual(self.test_unit_page.get_testunit_specification_type(), 'spec')
+            self.assertEqual(self.test_unit_page.get_spec_upper_limit(), str(random_upper_limit))
+            self.assertEqual(self.test_unit_page.get_spec_lower_limit(), str(random_lower_limit))
```

Comment:
```suggestion
        if spec_or_quan == 'spec':
            self.info('switch to quantification')
            self.test_unit_page.switch_from_spec_to_quan(lower_limit=random_lower_limit,
                                                         upper_limit=random_upper_limit)
        else:
            self.info('switch to specification')
            self.test_unit_page.switch_from_quan_to_spec(lower_limit=random_lower_limit,
                                                         upper_limit=random_upper_limit)
        self.info('refresh to make sure that data are updated successfully')
        self.base_selenium.refresh()
        self.assertEqual(self.test_unit_page.get_testunit_specification_type(), spec_or_quan)
        self.assertEqual(self.test_unit_page.get_spec_upper_limit(), str(random_upper_limit))
        self.assertEqual(self.test_unit_page.get_spec_lower_limit(), str(random_lower_limit))
```

Reply:
`self.assertEqual(self.test_unit_page.get_**spec**_upper_limit(), str(random_upper_limit))
` 

not the same as 



`self.assertEqual(self.test_unit_page.get_**quan_**upper_limit(), str(random_upper_limit))
` 